common/config/loader.go

```diff
@@ -83,9 +83,6 @@ func Load(env string, configDir string, zone string, config interface{}) error {
 		options = append(options, uconfig.File(f))
 	}
 
-	// expand env variables declared in .yaml files
-	options = append(options, uconfig.Expand(os.LookupEnv))
```

Comment:
This should improve things rather make it worse. A recent example: https://github.com/uber/cadence/issues/5923

Reply:
The fact that we do this $var expansion without proper documentation is not ideal. However there may be cadence configs in the wild that depends on this behavior. Dropping this at this point would break such cases. 
e.g. docker/config/bench/development.yaml was missed in this PR and there are more under host/testdata folder.

Just checked how this uconfig library works. It supports escaping with an additional `$`.
- "field: $ENV_VAR_ABC" -> resolves to "field: abc"
- "field: $$ENV_VAR_ABC" -> resolves to "field: $ENV_VAR_ABC"

So we might be fine with just documenting this behavior